client/electron/src/setup.ts

```diff
@@ -256,25 +254,41 @@ export class ElectronCapacitorApp {
     });
 
     this.MainWindow.on('close', (event) => {
-      if (this.forceClose) {
-        return;
-      }
-
-      const tray = this.isTrayEnabled();
-      if (tray) {
-        this.hideMainWindow();
+      if (process.platform === 'darwin') {
+        if (!this.macOSForceQuit) {
+          // Red X clicked
+          if (this.MainWindow.isFullScreen()) {
+            // If fullscreen, exit fullscreen first then hide window with a one-time listener
+            this.MainWindow.setFullScreen(false);
+            this.MainWindow.once('leave-full-screen', () => {
```

Comment:
Shouldn't it be declared prior to exiting fullscreen? I don't know the internal workings but I wouldn't be surprised if the event was fired right when the `setFullScreen(false)` is executed, and thus not triggering the `once('leave-full-screen')` event. And even if it seems to work right now, it's probably better to move it before just to be sure.